modules/launcher/src/test/java/org/mule/module/launcher/DeploymentServiceTestCase.java

```diff
@@ -1155,27 +1159,46 @@ public void receivesMuleContextDeploymentNotifications() throws Exception
     @Test
     public void undeploysStoppedApp() throws Exception
     {
-        addPackedAppFromBuilder(emptyAppFileBuilder);
-
-        deploymentService.start();
-
-        assertApplicationDeploymentSuccess(applicationDeploymentListener, emptyAppFileBuilder.getId());
-        final Application app = findApp(emptyAppFileBuilder.getId(), 1);
+        final Application app = deployApplication();
         app.stop();
         assertStatus(app, STOPPED);
 
         deploymentService.undeploy(app);
     }
 
     @Test
-    public void undeploysApplicationRemovingAnchorFile() throws Exception
+    public void whenAppIsStoppedStateIsPersistedAsDeploymentProperty() throws Exception
     {
-        addPackedAppFromBuilder(emptyAppFileBuilder);
+        Application app = deployApplication();
+        app.stop();
 
-        deploymentService.start();
+        DefaultMuleContext defaultMuleContext = (DefaultMuleContext) app.getMuleContext();
+        assertThat(defaultMuleContext.getRegistry().lookupObject(ARTIFACT_STOPPED_LISTENER), is(notNullValue()));
 
-        assertApplicationDeploymentSuccess(applicationDeploymentListener, emptyAppFileBuilder.getId());
-        Application app = findApp(emptyAppFileBuilder.getId(), 1);
+        Optional<Properties> properties = absent();
+        Properties deploymentProperties = resolveDeploymentProperties(emptyAppFileBuilder.getId(), properties);
+        assertThat(deploymentProperties.get(START_ARTIFACT_ON_DEPLOYMENT_PROPERTY), is(notNullValue()));
+        assertThat(deploymentProperties.get(START_ARTIFACT_ON_DEPLOYMENT_PROPERTY), Is.<Object>is("false"));
```

Comment:
Doesn't it work to use the statically imported "is"?
assertThat(deploymentProperties.get(START_ARTIFACT_ON_DEPLOYMENT_PROPERTY), is("false"));

Reply:
Apparently I can't statically import it because deploymentProperties.get(START_ARTIFACT_ON_DEPLOYMENT_PROPERTY) returns a dynamically defined value. Another option could be: `assertThat((String) deploymentProperties.get(START_ARTIFACT_ON_DEPLOYMENT_PROPERTY), is("false"));` but I'm worried this could lead to casting exceptions instead of assertion exceptions. Should I change it anyway or leave it as it is?